From b09caa693a3006d07190b55ba3a3d8cfac4fe364 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@nxp.com>
Date: Tue, 14 Aug 2018 08:05:12 +0800
Subject: [PATCH 4414/5242] MLK-19098 ARM: dts: imx7ulp-evk: use OTG ID
 function instead of GPIO

commit  2911e974c84526b108428dcdbb6deac00fde1165 from
https://source.codeaurora.org/external/imx/linux-imx.git

Since IMX7ULP_PAD_PTC13 has OTG ID function, so change the iomux
to be OTG ID instead of GPIO, with that we can use usb ID event
directly for role switch, which is more quick than extcon driver.
which also help on pass the type-C compliance test: "TD 4.5.6 DRP
connect Try.SNK DRP Test"

Acked-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7ulp-evk.dts |   21 +++++++--------------
 1 file changed, 7 insertions(+), 14 deletions(-)

diff --git a/arch/arm/boot/dts/imx7ulp-evk.dts b/arch/arm/boot/dts/imx7ulp-evk.dts
index 4af9fea..d7abcfb 100644
--- a/arch/arm/boot/dts/imx7ulp-evk.dts
+++ b/arch/arm/boot/dts/imx7ulp-evk.dts
@@ -61,8 +61,8 @@
 			compatible = "regulator-fixed";
 			reg = <0>;
 			pinctrl-names = "default", "sleep";
-			pinctrl-0 = <&pinctrl_usb_otg1>;
-			pinctrl-1 = <&pinctrl_usb_otg1>;
+			pinctrl-0 = <&pinctrl_usbotg1_vbus>;
+			pinctrl-1 = <&pinctrl_usbotg1_vbus>;
 			regulator-name = "usb_otg1_vbus";
 			regulator-min-microvolt = <5000000>;
 			regulator-max-microvolt = <5000000>;
@@ -92,14 +92,6 @@
 		};
 	};
 
-	extcon_usb1: extcon_usb1 {
-		compatible = "linux,extcon-usb-gpio";
-		id-gpio = <&gpio_ptc 13 GPIO_ACTIVE_HIGH>;
-		pinctrl-names = "default", "sleep";
-		pinctrl-0 = <&pinctrl_extcon_usb1>;
-		pinctrl-1 = <&pinctrl_extcon_usb1>;
-	};
-
 	pf1550-rpmsg {
 		compatible = "fsl,pf1550-rpmsg";
 		sw1_reg: SW1 {
@@ -323,15 +315,15 @@
 			>;
 		};
 
-		pinctrl_usb_otg1: usbotg1grp {
+		pinctrl_usbotg1_vbus: otg1vbusgrp {
 			fsl,pins = <
 				IMX7ULP_PAD_PTC0__PTC0		0x20000
 			>;
 		};
 
-		pinctrl_extcon_usb1: extcon1grp {
+		pinctrl_usbotg1_id: otg1idgrp {
 			fsl,pins = <
-				IMX7ULP_PAD_PTC13__PTC13	0x10003
+				IMX7ULP_PAD_PTC13__USB0_ID	0x10003
 			>;
 		};
 
@@ -531,7 +523,8 @@
 
 &usbotg1 {
 	vbus-supply = <&reg_usb_otg1_vbus>;
-	extcon = <0>, <&extcon_usb1>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usbotg1_id>;
 	srp-disable;
 	hnp-disable;
 	adp-disable;
-- 
1.7.9.5

